### DISCUSSION TITLE
Debug: Z-fighting debug window

### DISCUSSION THREAD

**Discussion by karst** [OpenRCT2/OpenRCT2#18330] [Feature requests]
Perfect for fixing bounding box issues, offset sprites and adding new track pieces. I'd like to see the introduction of a 3 by 3 number window that makes can change parameters on the fly without having to continuously build the game over and over when changing a parameter. This will take trial and error away from the bug fixes and will make workflow overall a lot faster.

Instead of filling in a number you can type in `{`windowAx`, `windowAy`, height + `windowAz`}`, etc in the code instead, which will be linked to a spinner in a new debug window with the following layout.

|   |  X | Y  | Z |
|---|---|---|---|
| Sprite location |  `windowAx` ↕| `windowAy` ↕| `windowAz` ↕| 
| Bounding Box Volume |  `windowBx` ↕| `windowBy` ↕| `windowBz` ↕|
|  Bouding Box offset | `windowCx` ↕| `windowCy` ↕| `windowCz` ↕|

I have no clue how to code a window, which is why I am requesting it. 

Additionally it'd be nice to have the visible bounding box toggle available on this window as well.

**Comment by ocalhoun6** [comment]
Alternatively, maybe these parameters could be exposed to the plugin system, and then a plugin could be made for it?

That might even end up being useful to people who make custom parks and want to really fine-tune the positions of things. Or to help combat the z-fighting issues that often occur when placing many objects into an overlapping space with clearance checks disabled.

**Comment by karst** [comment]
@Broxzier Wasn't it you that said you could link spinners to the coords (linked to a cheat window spinner orso)? Or was it someone else? I vaguely remember someone saying it when fixing bounding boxes. When talking to Michael while waiting in line for Untamed he said it be way too performance intensive to be a viable option. Though if it would be possible it would be a massive time saver.

**Comment by spacek531** [comment]
The drawing code modifications that @Gymnasiast wants to make will have to come first to unify track drawing code so that its values can reasonably be intercepted in this manner.
